test/nl/surf/eduhub_rio_mapper/worker_test.clj

```diff
@@ -149,38 +153,56 @@
-    (worker/purge! config)
-
-    ;; queue a successful job
-    (worker/enqueue! config {:queue "foo"})
-    (is (= {:job {:queue "foo"}, :status :pending, :data nil}
-           @last-seen-status))
-
-    ;; spin up a bunch of workers
-    (let [workers (mapv (fn [_] (worker/start-worker! config)) (range num-of-workers))]
-
-      ;; wait for successful job to finish
-      (wait-for-expected {:job    {:queue "foo"}
-                          :status :done
-                          :data   {:queue "foo"}}
-                         last-seen-status 10)
-
-      ;; queue and wait for error job
-      (worker/enqueue! config {:queue "foo", :error? true})
-      (wait-for-expected {:job    {:queue "foo", :error? true}
-                          :status :error
-                          :data   {:queue  "foo", :error? true}}
-                         last-seen-status 10)
-
-      ;; stop workers
-      (doseq [worker workers] (worker/stop-worker! worker))
-      (worker/purge! config))))
+  (testing "enqueuing jobs"
+    (let [last-seen-status (atom nil)
+          config           (-> config
+                               (assoc-in [:worker :error-fn] :error?)
+                               (assoc-in [:worker :run-job-fn] identity)
+                               (assoc-in [:worker :set-status-fn]
+                                         (fn [job status & [data]]
+                                           (reset! last-seen-status {:job    job
+                                                                     :status status
+                                                                     :data   data}))))]
+      (worker/purge! config)
+
+      ;; queue a successful job
+      (worker/enqueue! config {:queue "foo"})
+      (is (= {:job {:queue "foo"}, :status :pending, :data nil}
+             @last-seen-status))
+
+      ;; spin up a bunch of workers
+      (let [workers (mapv (fn [_] (worker/start-worker! config)) (range num-of-workers))]
+
+        ;; wait for successful job to finish
+        (wait-for-expected {:job    {:queue "foo"}
+                            :status :done
+                            :data   {:queue "foo"}}
+                           last-seen-status 10)
+
+        ;; queue and wait for error job
+        (worker/enqueue! config {:queue "foo", :error? true})
+        (wait-for-expected {:job    {:queue "foo", :error? true}
+                            :status :error
+                            :data   {:queue "foo", :error? true}}
+                           last-seen-status 10)
+
+        ;; stop workers
+        (doseq [worker workers] (worker/stop-worker! worker))
+        (worker/purge! config)))
+
+    (testing "webhook"
+      (let [request       (atom nil)
+            set-status-fn (cli/make-set-status-fn config)
+            mock-webhook  (fn mock-webhook [req]
+                            (swap! request (constantly req))
```

Comment:
`(reset! last-seen-request-atom req)`

Reply:
Ah, of course there's a function for that!